esp8266/esp_mphal.c

```diff
@@ -149,6 +151,29 @@ void mp_hal_signal_input(void) {
     #endif
 }
 
+void mp_hal_uart_rx_intr(int uart_no) {
+    int ch;
+    mp_obj_t term = MP_STATE_PORT(term_obj);
+    bool uart_term=(term == NULL || term == MP_STATE_PORT(pyb_uart_objs)[uart_no]);
```

Comment:
should this be 

bool uart_term = (term == NULL || term == MP_STATE_PORT(pyb_uart_objs)[uart_no]);